src/Controls/src/Core/Menu/MenuFlyoutSubItem.cs

```diff
@@ -87,7 +82,7 @@ public void RemoveAt(int index)
 		{
 			var item = _menus[index];
 			RemoveLogicalChild((Element)item, index);
-			NotifyHandler(nameof(IMenuFlyoutHandler.Remove), index, item);
+			NotifyHandler(nameof(IMenuFlyoutHandler.Remove), index, item as IMenuElement);
```

Comment:
I don't know if this will work...

I feel like we've hit a point on this one where we actually need to maintain two different internal lists. 

`_menus` no longer represents `_menus` it's a mix of types. 

If someone calls `RemoveAt(index)` they are going to expect that it removes the menu specifically at the given index (we should also make sure this removes the ImageSource added)

So, I think we have to keep `_menus` as a list of IMenuElement to satisfy all the `IList<IMenuElement>` implementations. 

And then the backing store for logical children basically has to be a mix of both. 

We basically do this with `Layout.cs`, it has an internal list of `_children` it uses to satisfy the direct children but then it adds each of this to the logical children on the parent